cmd/zapi/cmd/new/new.go

```diff
@@ -16,20 +17,27 @@ var New = &charm.Spec{
 	Short: "create a new space",
 	Long: `The new command takes a single argument and creates a new, empty space
 named as specified.`,
-	New: func(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
-		return &Command{Command: parent.(*cmd.Command)}, nil
-	},
+	New: NewFn,
 }
 
 func init() {
 	cmd.CLI.Add(New)
 }
 
-type Command struct {
+type NewCommand struct {
 	*cmd.Command
+	kind     string
+	datapath string
 }
 
-func (c *Command) Run(args []string) error {
+func NewFn(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &NewCommand{Command: parent.(*cmd.Command)}
+	f.StringVar(&c.kind, "k", string(storage.FileStore), "kind of storage for this space")
```

Comment:
Consider having `storage.Kind` implement `flag.Value` so you can return an error if an invalid option is specified.